### ISSUE TITLE
Unsupported page encoding when reading Parquet

### ISSUE THREAD

**Issue by jaldaadithya** [duckdb/duckdb#3981]
Hi Team, We are unable to read some of the parquet files like aws2.parquet and complex.parquet from this repository

#### What happens?
When we trigger a select query such as select * from aws2.parquet limit 5, we are able to see **Unsupported page encoding when reading Parquet**

#### To Reproduce
1) Download the aws2.parquet file from https://github.com/duckdb/duckdb/blob/master/data/parquet-testing/aws2.parquet
2) Go to https://shell.duckdb.org/
3) Execute the **.files add** command to select the downloaded aws2.parquet file
4) Execute the **.files list** command to see the list of files uploaded (at present we only uploaded aws2.parquet)
5) Execute a select command **SELECT * FROM 'aws2.parquet' limit 2;**
6) Now we are able to see the exception Unsupported page encoding when reading Parquet
![image](https://user-images.githubusercontent.com/19626378/176153949-fcdd15ba-a3fc-4e17-8b10-6e413f44c6e2.png)

#### Environment (please complete the following information):
 - OS: Mac OS 12.4 (21F79) - Mac M1 (in my local setup for JDBC) and tried in Chrome DuckDB shell
 - DuckDB Version: We are using https://shell.duckdb.org/ and in my local setup I'm was 0.4.0
 - DuckDB Client: Shell wasm and JDBC (In my local)

#### Identity Disclosure:
 - Full Name: Adithya Jalda

Team, We are getting this exception when we perform the same steps in the JDBC client as well, But when we tried parsing with Java code using Apache parquet Reader it's working fine
We tried aws2.parquet, complex.parquet from this location https://github.com/duckdb/duckdb/blob/master/data/parquet-testing/

**Comment by Mytherin** [comment]
Thanks for the report! I can confirm this seems to be a problem in the Parquet reader.

**Comment by hannes** [comment]
There was a branch at some point that could read those...

**Comment by jaldaadithya** [comment]
Team, I tried almost all branches supported by my Mac M1, But all Java Jdbc client jars gave me the same error output,
@hannes could you please provide us with which branch the fix was present.

**Comment by Mytherin** [comment]
> There was a branch at some point that could read those...

The code that can read them is actually there but commented out. It appears to be correctly reading the data, it just needs to be turned into a (streaming) column reader.

**Comment by jaldaadithya** [comment]
Thank you @Mytherin, please let us know when the code is updated, I'll test it out with different parquet files

**Comment by aryounce** [comment]
This has been an issue for me as well. I checked the Parquet extension and the supported page encodings are limited. I worked around the problem by only using DuckDB supported encodings in my Parquet ETL process.

Per the latest Parquet extension source: https://github.com/duckdb/duckdb/blob/688abf6324397108fcc9e6f2dd388d04c0910586/extension/parquet/column_reader.cpp#L284

The only encodings currently supported are [Plain](https://github.com/apache/parquet-format/blob/master/Encodings.md#plain-plain--0), [Plain/RLE Dictionary](https://github.com/apache/parquet-format/blob/master/Encodings.md#dictionary-encoding-plain_dictionary--2-and-rle_dictionary--8), and [Delta](https://github.com/apache/parquet-format/blob/master/Encodings.md#delta-encoding-delta_binary_packed--5). Previously I was using a few others more appropriate for my data types and encountered the reported error. Note that DuckDB seems to read the metadata for columns with unsupported page encodings without issue.

**Comment by tobilg** [comment]
I get the same error when trying to read Parquet files generated by AWS Kinesis Firehose DeliveryStreams.

```bash
select * from read_parquet(['/Users/myuser/Downloads/test.parquet']);
Error: Invalid Error: Unsupported page encoding
```

Not sure if one can influence the encodings in the write process, as the only configuration options are outlined in https://docs.amazonaws.cn/en_us/AWSCloudFormation/latest/UserGuide/aws-properties-kinesisfirehose-deliverystream-parquetserde.html

You can check the file at https://cdn.modestats.com/incoming/test.parquet if you like. If I use `parquet-tools csv test.parquet`, I can see the content without any problems for example.

**Comment by tobilg** [comment]
As an update to my former comment, I switched to the following `OutputConfiguration` for my Firehose DeliveryStream, and it works:

```yaml
OutputFormatConfiguration: 
  Serializer: 
    ParquetSerDe: 
      Compression: SNAPPY
      WriterVersion: V1
```

I suspect it's the `EnableDictionaryCompression` that makes the difference.

**Comment by Mytherin** [comment]
Thanks for the report! Looks like this file uses the `DELTA_BYTE_ARRAY` encoding. I will have a look to see if we can support this.

**Comment by Mytherin** [comment]
Should all be implemented on #5457 now

**Comment by tobilg** [comment]
Thanks a lot @hannes / @Mytherin!
